adapters/lora/src/main/java/org/eclipse/hono/adapter/lora/LoraCommandSubscriptions.java

```diff
@@ -0,0 +1,178 @@
+     */
+    public void init() {
+        NotificationEventBusSupport.registerConsumer(vertx, TenantChangeNotification.TYPE,
+                notification -> {
+                    if (LifecycleChange.DELETE.equals(notification.getChange())
+                            || (LifecycleChange.UPDATE.equals(notification.getChange())
+                                    && !notification.isTenantEnabled())) {
+                        closeConsumersForTenant(notification.getTenantId());
+                    }
+                });
+
+        NotificationEventBusSupport.registerConsumer(vertx, DeviceChangeNotification.TYPE,
+                notification -> {
+                    if (LifecycleChange.DELETE.equals(notification.getChange())) {
+                        closeConsumersForGateway(notification.getTenantId(), notification.getDeviceId());
+                    }
+                });
+    }
+
+    @SuppressWarnings("rawtypes")
+    private void closeConsumersForGateway(final String tenantId, final String gatewayId) {
+        if (!commandSubscriptions.entrySet().stream().anyMatch(e -> e.getKey().getDeviceId().equals(gatewayId))) {
+            return;
+        }
+
+        final Span span = TracingHelper
+                .buildSpan(tracer, null, "gateway deleted, corresponding command consumer will be closed\"",
+                        getClass().getSimpleName())
+                .withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_CLIENT)
+                .start();
+        TracingHelper.setDeviceTags(span, tenantId, gatewayId);
+        final SubscriptionKey key = new SubscriptionKey(tenantId, gatewayId);
+        final TriTuple<CommandConsumer, LoraProvider, Context> subscription = commandSubscriptions.get(key);
+        final List<Future> consumerCloseFutures = Collections.synchronizedList(new ArrayList<>());
+        final Context ctx = subscription.three();
+        ctx.runOnContext(x -> {
+            final CommandConsumer commandConsumer = subscription.one();
+            consumerCloseFutures.add(commandConsumer.close(null));
+        });
+        CompositeFuture.join(consumerCloseFutures)
+                .onComplete(x -> commandSubscriptions.remove(key))
+                .onComplete(x -> span.finish());
+    }
+
+    @SuppressWarnings("rawtypes")
+    private void closeConsumersForTenant(final String tenantId) {
+        if (!commandSubscriptions.entrySet().stream().anyMatch(e -> e.getKey().getTenant().equals(tenantId))) {
+            return;
+        }
+
+        final Span span = TracingHelper
+                .buildSpan(tracer, null, "check command subscriptions on tenant disabled/deleted event",
+                        getClass().getSimpleName())
+                .withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_CLIENT)
+                .start();
+        TracingHelper.setDeviceTags(span, tenantId, null);
+        span.log("tenant disabled/deleted, corresponding command consumers will be closed");
```

Comment:
How about changing the span name to `close command subscriptions for disabled/deleted tenant`?
The span log message can be omitted then.